src/function/matrix/subset.js

```diff
@@ -73,29 +84,72 @@ export const createSubset = /* #__PURE__ */ factory(name, dependencies, ({ typed
 
     // set subset
     'Array, Index, any': function (value, index, replacement) {
+      if (isEmptyIndex(index)) {
+        return value
+      }
+      validateIndexSourceSize(value, index)
       return matrix(clone(value))
-        .subset(index, replacement, undefined)
+        .subset(index, _broadcastReplacement(replacement, index), undefined)
         .valueOf()
     },
 
     'Array, Index, any, any': function (value, index, replacement, defaultValue) {
+      if (isEmptyIndex(index)) {
+        return value
+      }
+      validateIndexSourceSize(value, index)
       return matrix(clone(value))
-        .subset(index, replacement, defaultValue)
+        .subset(index, _broadcastReplacement(replacement, index), defaultValue)
         .valueOf()
     },
 
     'Matrix, Index, any': function (value, index, replacement) {
-      return value.clone().subset(index, replacement)
+      if (isEmptyIndex(index)) {
+        return value
+      }
+      validateIndexSourceSize(value, index)
+      return value.clone().subset(index, _broadcastReplacement(replacement, index))
     },
 
     'Matrix, Index, any, any': function (value, index, replacement, defaultValue) {
-      return value.clone().subset(index, replacement, defaultValue)
+      if (isEmptyIndex(index)) {
+        return value
+      }
+      validateIndexSourceSize(value, index)
+      return value.clone().subset(index, _broadcastReplacement(replacement, index), defaultValue)
     },
 
```

Comment:
There's a lot of new code duplication in these various different signatures. Could this be converted into one implementation that does the work and all the rest forwarding to it, to avoid this duplication and ease future maintenance?

Reply:
Yes I see what you mean.

Maybe `'Array, Index, any'` can redirect to `'Array, Index, any, any'` (using `typed.referToSelf`), and similarly for `'Matrix, Index, any'`. What do you think @dvd101x ?